### PR TITLE
[sdb] Add support for reading the sourcelink data from portable pdb files using a new SourceLink property on ModuleMirror objects.

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/mono/mono/issues/10038.

### CODE REVIEW

File: mono/metadata/debug-mono-ppdb.c [hunk 1]
```diff
@@ -748,3 +756,18 @@ mono_ppdb_lookup_method_async_debug_info (MonoDebugMethodInfo *minfo)
...
 	return res;
 }
+
+char*
+mono_ppdb_get_sourcelink (MonoDebugHandle *handle)
+{
+	MonoPPDBFile *ppdb = handle->ppdb;
+	MonoImage *image = ppdb->image;
+
+	guint8 sourcelink_guid [16] = { 0x56, 0x05, 0x11, 0xCC, 0x91, 0xA0, 0x38, 0x4D, 0x9F, 0xEC, 0x25, 0xAB, 0x9A, 0x35, 0x1A, 0x6A };
```
Comment (jaykrell) [inline mono/metadata/debug-mono-ppdb.c]: static const?

File: mono/metadata/debug-mono-ppdb.c [hunk 2]
```diff
@@ -657,7 +664,8 @@ table_locator (const void *a, const void *b)
 }
 
 static gboolean
-compare_guid (guint8* guid1, guint8* guid2) {
+compare_guid (guint8* guid1, guint8* guid2)
```
Comment (jaykrell) [inline mono/metadata/debug-mono-ppdb.c]: const? Could use memcmp but ok either way.

**Comment by luhenry** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
@monojenkins build Linux AArch64 FullAOT+LLVM

**Comment by luhenry** [comment]
It seems `mcs` doesn't support this parameter. We should disable the test when compiling with it /cc @marek-safar

**Comment by marek-safar** [comment]
I fixed the mcs but compile bcl tests is not important in this step and will remove it as well

**Comment by marek-safar** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@vargaz could you please address the new tests failures

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
The windows failures are now fixed.

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Pull request merged**